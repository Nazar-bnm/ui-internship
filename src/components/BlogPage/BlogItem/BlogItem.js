import React from 'react';

import './BlogItem.scss';

const CN = 'blog-item';

const BlogItem = ({
  title,
  date,
  photo,
  text,
  labels,
  className
}) => (
  <div className={`${CN} ${className}`}>
    <h3 className={`${CN}__heading`}>{title}</h3>
    <span className={`${CN}__date`}>{date}</span>
    <div className={`${CN}__image-wrapper`}>
      <img
        className={`${CN}__image`}
        alt="Publication"
        src={photo}
      />
      <div className="onHover">{labels.join(', ')}</div>
    </div>
    <p className={`${CN}__text`}>{text}</p>
  </div>
);

export default BlogItem;
