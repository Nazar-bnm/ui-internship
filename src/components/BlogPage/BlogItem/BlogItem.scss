@import '../../../scss/global_styles';

.blog-item {
  margin: 10px 0;
  padding: 10px;

  &__heading, &__date {
    margin: 10px 0;
    text-transform: uppercase;
  }

  &__date, &__text {
    font-size: 0.8rem;
    font-family: font(primary-light);
  }

  &__text {
    text-align: justify;
    color: $dove-gray;
  }

  &__image {
    width: 100%;

    &-wrapper {
      @include flex;
      position: relative;
      margin: 10px 0;
    }
  }

  .onHover {
    @include flex($justify: center, $align: center);
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    padding: 20px;
    font-size: 1.8rem;
    text-align: center;
    font-family: font(primary-light);
    text-transform: uppercase;
    color: transparent;

    &:hover {
      color: $white;
      background-image:
      linear-gradient(
        to bottom right, 
        $orange-transparent, $yellow-transparent
      );
    }
  }
}

.bottom-border {
  padding-bottom: 30px;
  border-bottom: 3px solid $black;
}
