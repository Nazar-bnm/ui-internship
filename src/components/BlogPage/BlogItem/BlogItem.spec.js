import React from 'react';
import { shallow } from 'enzyme';

import BlogItem from './BlogItem';

describe('<BlogItem />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<BlogItem 
      key='1'
      title='Title'
      date='11 August 2019'
      photo='https://res.cloudinary.com/dr8z1kafr/image/upload/v1573482788/Interns/blog/1.jpg'
      text='Text'
      labels={['summer', 'fun']}
      className='bottom-border'
    />);
    expect(wrapper).toMatchSnapshot();
  });
});