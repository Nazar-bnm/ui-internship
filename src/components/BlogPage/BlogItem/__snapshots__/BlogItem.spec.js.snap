// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogItem /> should match snapshot 1`] = `
<div
  className="blog-item bottom-border"
>
  <h3
    className="blog-item__heading"
  >
    Title
  </h3>
  <span
    className="blog-item__date"
  >
    11 August 2019
  </span>
  <div
    className="blog-item__image-wrapper"
  >
    <img
      alt="Publication"
      className="blog-item__image"
      src="https://res.cloudinary.com/dr8z1kafr/image/upload/v1573482788/Interns/blog/1.jpg"
    />
    <div
      className="onHover"
    >
      summer, fun
    </div>
  </div>
  <p
    className="blog-item__text"
  >
    Text
  </p>
</div>
`;
